area/area.go

```diff
@@ -0,0 +1,89 @@
+package area
+
+import (
+	"time"
+
+	"github.com/almighty/almighty-core/errors"
+	"github.com/almighty/almighty-core/gormsupport"
+	"github.com/goadesign/goa"
+	"github.com/jinzhu/gorm"
+	uuid "github.com/satori/go.uuid"
+	"golang.org/x/net/context"
+)
+
+const APIStringTypeAreas = "areas"
+
+// Area describes a single Area
+type Area struct {
+	gormsupport.Lifecycle
+	ID      uuid.UUID `sql:"type:uuid default uuid_generate_v4()" gorm:"primary_key"` // This is the ID PK field
+	SpaceID uuid.UUID `sql:"type:uuid"`
+	Path    string
+	Name    string
+	Version int
+}
+
+// TableName overrides the table name settings in Gorm to force a specific table name
+// in the database.
+func (m *Area) TableName() string {
+	return "areas"
+}
+
+// Repository describes interactions with Areas
+type Repository interface {
+	Create(ctx context.Context, u *Area) error
+	List(ctx context.Context, spaceID uuid.UUID) ([]*Area, error)
+	Load(ctx context.Context, id uuid.UUID) (*Area, error)
+}
+
+// NewAreaRepository creates a new storage type.
+func NewAreaRepository(db *gorm.DB) Repository {
+	return &GormAreaRepository{db: db}
+}
+
+// GormAreaRepository is the implementation of the storage interface for Areas.
+type GormAreaRepository struct {
+	db *gorm.DB
+}
+
+// Create creates a new record.
+func (m *GormAreaRepository) Create(ctx context.Context, u *Area) error {
+	defer goa.MeasureSince([]string{"goa", "db", "area", "create"}, time.Now())
+
+	u.ID = uuid.NewV4()
+
```

Comment:
Thanks for the feedback @pranavgore09 - yes we could, but the path is the parent path - the name has been renamed at the service level to parent_path..